conan/tools/cmake/presets.py

```diff
@@ -126,13 +123,23 @@ def write_cmake_presets(conanfile, toolchain_file, generator):
                 data = {"version": 4, "include": [preset_path]}
             else:
                 data = json.loads(load(user_presets_path))
+                data = _clean_deleted_presets(data)
                 if preset_path not in data["include"]:
                     data["include"].append(preset_path)
 
             data = json.dumps(data, indent=4)
             save(user_presets_path, data)
 
 
+def _clean_deleted_presets(data):
+    includes = []
+    for include in data["include"]:
+        if os.path.exists(include):
+            includes.append(include)
+    data["include"] = includes
+    return data
```

Comment:
```suggestion
def _clean_deleted_presets(data):
    data["include"] = [i for i in data["include"] if os.path.exists(i)]
```

And as it is called in a single place, maybe just use this single line instead